lib/linter/apply-disable-directives.js

```diff
@@ -456,10 +463,31 @@ module.exports = ({ directives, disableFixes, problems, reportUnusedDisableDirec
         reportUnusedDisableDirectives
     });
 
-    return reportUnusedDisableDirectives !== "off"
-        ? lineDirectivesResult.problems
-            .concat(blockDirectivesResult.unusedDirectives)
+    if (reportUnusedDisableDirectives !== "off") {
+        const rulesToIgnore = configuredRules && ruleFilter
+            ? new Set(Object.keys(configuredRules).filter(ruleId => {
+                const severity = ConfigOps.getRuleSeverity(configuredRules[ruleId]);
+
+                return !ruleFilter({ severity, ruleId });
+            }))
+            : new Set();
+
+        const unusedDirectives = blockDirectivesResult.unusedDirectives
             .concat(lineDirectivesResult.unusedDirectives)
-            .sort(compareLocations)
-        : lineDirectivesResult.problems;
+            .filter(unusedDirective => {
+                const ruleId = unusedDirective.ruleId;
+
+                if (ruleId === null && rulesToIgnore.size > 0 || rulesToIgnore.has(ruleId)) {
```

Comment:
It seems like we could simplify this by adding a `null` to `rulesToIgnore` and then just checking `rulesToIgnore.has(ruleId)`?

Also, a comment here explaining why we are filtering out rules without a `ruleId` would be helpful.